File: backend/app/schemas/diagnostic_code.py
"""
Pydantic schemas for Diagnostic Code.
"""
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, Field


class DiagnosticCodeBase(BaseModel):
    """Base schema for Diagnostic Code."""
    
    code: str = Field(..., min_length=1, max_length=50, description="Diagnostic code")
    description: str = Field(..., min_length=1, description="Code description")
    category: Optional[str] = Field(None, max_length=100, description="Code category")
    subcategory: Optional[str] = Field(None, max_length=100, description="Code subcategory")
    severity: Optional[str] = Field(None, description="Severity level")
    is_active: bool = Field(True, description="Whether the code is active")


class DiagnosticCodeCreate(DiagnosticCodeBase):
    """Schema for creating a new Diagnostic Code."""
    pass


class DiagnosticCodeUpdate(BaseModel):
    """Schema for updating a Diagnostic Code."""
    
    description: Optional[str] = None
    category: Optional[str] = None
    subcategory: Optional[str] = None
    severity: Optional[str] = None
    is_active: Optional[bool] = None


class DiagnosticCodeResponse(DiagnosticCodeBase):
    """Schema for Diagnostic Code response."""
    
    id: int
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class DiagnosticCodeList(BaseModel):
    """Schema for list of Diagnostic Codes."""
    
    total: int
    items: List[DiagnosticCodeResponse]
    skip: int
    limit: int

File: frontend/src/types/diagnosticCode.ts
import { z } from 'zod'

export const diagnosticCodeSchema = z.object({
  id: z.number(),
  code: z.string(),
  description: z.string(),
  category: z.string().nullable(),
  subcategory: z.string().nullable(),
  severity: z.string().nullable(),
  is_active: z.boolean(),
  created_at: z.string(),
  updated_at: z.string(),
})

export const diagnosticCodeCreateSchema = z.object({
  code: z.string().min(1, 'Code is required'),
  description: z.string().min(1, 'Description is required'),
  category: z.string().optional(),
  subcategory: z.string().optional(),
  severity: z.enum(['low', 'medium', 'high', 'critical']).optional(),
  is_active: z.boolean().default(true),
})

export const diagnosticCodeListSchema = z.object({
  total: z.number(),
  items: z.array(diagnosticCodeSchema),
  skip: z.number(),
  limit: z.number(),
})

export type DiagnosticCode = z.infer<typeof diagnosticCodeSchema>
export type DiagnosticCodeCreate = z.infer<typeof diagnosticCodeCreateSchema>
export type DiagnosticCodeList = z.infer<typeof diagnosticCodeListSchema>
